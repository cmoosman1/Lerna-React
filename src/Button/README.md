# Button component

## Usage
Import component with: `import Button from '@cmoosman/button';`
