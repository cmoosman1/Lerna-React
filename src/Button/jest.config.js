module.export = {
    roots: ['<rootDir>/src'],
    transform: {
      '\\.(js|jsx)?$': 'babel-jest',
    },
    moduleNameMapper: {
        '\\.(css|less)$': '<rootDir>/test/jest/__mocks__/styleMock.js',
    },
    jest: {
        "moduleNameMapper": {
          "\\.(css|scss)$": "<rootDir>/__mocks__/styleMock.js"
        }
    },
    testMatch: ['<rootDir>/src/**/>(*.)test.{js, jsx}'], // finds test
    moduleFileExtensions: ['js', 'jsx', 'json', 'node'],
    testPathIgnorePatterns: ['/node_modules/', '/public/'],
    setupFilesAfterEnv: [
      '@testing-library/jest-dom/extend-expect', 
      '@testing-library/react/cleanup-after-each'
    ] // setupFiles before the tests are ran
  };