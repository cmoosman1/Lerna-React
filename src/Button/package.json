{
  "name": "@cmoosman/button",
  "description": "Button component",
  "version": "1.0.8",
  "license": "MIT",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "cross-env webpack -p --config ../../webpack.config.js",
    "clean": "rimraf dist && rimraf node_modules && rimraf package-lock.json",
    "test": "jest",
    "test:watch": "jest --watch",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2"
  },
  "devDependencies": {
    "@babel/core": "^7.11.1",
    "@babel/plugin-proposal-class-properties": "^7.10.4",
    "@storybook/addon-actions": "^6.0.13",
    "@storybook/addon-essentials": "^6.0.13",
    "@storybook/addon-links": "^6.0.13",
    "@storybook/react": "^6.0.13",
    "@storybook/theming": "^6.0.13",
    "@testing-library/jest-dom": "^5.11.3",
    "@testing-library/react": "^10.4.8",
    "babel-loader": "^8.1.0",
    "cross-env": "^7.0.2",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^26.4.0",
    "react-is": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "webpack-cli": "^3.3.12"
  },
  "dependencies": {
    "@storybook/addon-a11y": "^6.0.13"
  }
}
