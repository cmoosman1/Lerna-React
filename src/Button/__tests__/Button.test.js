import React from 'react';
import Button from '../src/Button';
import renderer from 'react-test-renderer';
import { cleanup, render, fireEvent } from '@testing-library/react';

it('renders correctly', () => {
  const tree = renderer
    .create(<Button />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Button', () => {
    let getByTestId;
    let elem;
    let onClickHandler = jest.fn();
    afterEach(cleanup);
    beforeEach(() => {
        ({getByTestId} = render(<Button onClick={onClickHandler}/>));
        elem = getByTestId('base-button');
        fireEvent.click(getByTestId('base-button'));
    });

    it('check button render label', () => {
      expect(elem.innerHTML).toBe('Apply Now');
    });

    it('button onClick fired', () => {
        expect(onClickHandler).toHaveBeenCalled();
    });
});