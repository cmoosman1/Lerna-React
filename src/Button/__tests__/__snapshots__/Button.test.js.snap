// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="storybook-button storybook-button--medium storybook-button--secondary"
  data-testid="base-button"
  style={null}
  type="button"
>
  Apply Now
</div>
`;
